rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -152,33 +140,12 @@ public boolean isMember(FederationMember federationMember){
         return this.members.contains(federationMember);
     }
 
-    @Override
     public String toString() {
         return String.format("%d of %d signatures federation", getNumberOfSignaturesRequired(), members.size());
     }
 
-    @Override
-    public boolean equals(Object other) {
-        if (this == other) {
-            return true;
-        }
-
-        if (other == null || this.getClass() != other.getClass()) {
-            return false;
-        }
-
-        Federation otherFederation = (Federation) other;
-
-        return this.getNumberOfSignaturesRequired() == otherFederation.getNumberOfSignaturesRequired() &&
-            this.getSize() == otherFederation.getSize() &&
-            this.getCreationTime().equals(otherFederation.getCreationTime()) &&
-            this.creationBlockNumber == otherFederation.creationBlockNumber &&
-            this.btcParams.equals(otherFederation.btcParams) &&
-            this.members.equals(otherFederation.members) &&
-            this.getRedeemScript().equals(otherFederation.getRedeemScript());
-    }
+    public abstract boolean equals(Object other);
 
-    @Override
     public int hashCode() {
```

Comment:
Maybe this should be part of StandarFederation, right? Because ERPFederation is defining it's own hashCode

Reply:
Yes, of course. Absolutely right

Reply:
Yes, sounds good.